public/app/features/variables/inspect/utils.ts

```diff
@@ -419,3 +367,24 @@ export function flattenPanels(panels: PanelModel[]): PanelModel[] {
 
   return result;
 }
+
+// Accepts an array of panel models, and returns an array of panel IDs paired with
+// the names of any template variables found
+export function getPanelVars(panels: PanelModel[]) {
+  const panelsByVar: Record<string, Set<number>> = {};
+  for (const p of panels) {
+    const jsonString = stringifyPanelModel(p);
+    const repeats = [...jsonString.matchAll(/"repeat":"([^"]+)"/g)].map((m) => m[1]);
+    const varRegexMatches = jsonString.match(variableRegex)?.map((m) => getVariableName(m)) ?? [];
```

Comment:
🥴  really wish we had a better way to do this. Stringifying and then doing regex on the result is a very cludgy way to go about this 😅 

Not suggesting we do this now, but i wonder if it's faster to iterate over all properties looking for variables, than converting to JSON and regex-ing?

Reply:
I agree that it's not ideal, but I didn't want to meddle with that aspect of the variables pipeline with this PR 😋.
Definitely something to look into in the future though